### ISSUE TITLE
Enhancement: [_animation.scss] add reduced motion support for .calcite-animate global CSS class

### ISSUE THREAD

**Issue by caripizza** [Esri/calcite-components#3489]
### Description
Add [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) CSS media feature for the [`.calcite-animate`](https://github.com/Esri/calcite-components/blob/master/src/assets/styles/_animation.scss#L48) global CSS class. This will ensure animations only apply when the prefers-reduced-motion media feature matches `no-preference` on a system (ie., the class will only animate if a user hasn't enabled "Reduce motion" on their system).

Proposed change (courtesy [@tim-white-esri](https://github.com/Esri/calcite-components/issues/3081#issuecomment-960101416)):
```scss
// because default state is opacity: 0; need to make sure that is explicitly set to 1
@media (prefers-reduced-motion: reduce) {
    .calcite-animate {
        opacity: 1;
        animation: none;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .calcite-animate {
        opacity: 0;
        animation-fill-mode: both;
        animation-duration: var(--calcite-animation-timing);
    }
}

```

### Acceptance Criteria 
The src/assets/styles/_animation.scss global stylesheet includes the new CSS media features (above snippet) for the .calcite-animate class.

### Relevant Info 
See pr #2607 related file changes for helpful information on using the global class.

#### Which Component
n/a (global class)

#### Example Use Case
n/a

**Comment by tim-white-esri** [comment]
Any traction on this? I just received another complaint from a site visitor about animations not respecting reduced motion. 

This is important for accessibility.

**Comment by geospatialem** [comment]
@tim-white-esri Thanks for re-pinging this issue. We'll be adding this to our upcoming sprint. For a workaround [Codepen](https://codepen.io/jcfranco/pen/JjvjpOV) in the meantime.

**Comment by geospatialem** [comment]
The solution will update the internal duration factor based on user preferences. This will still give developers the flexibility to override the animation, but would not be supported by Calcite.

**Comment by geospatialem** [comment]
Experiencing some unexpected behavior in Windows with the `loader` component in `beta.95` when animations are disabled. The expectation is no animation is present when users set the system preferences to disable animations.

Rerouting back to dev, @anveshmekala where [this line](https://github.com/Esri/calcite-components/pull/5314/files#diff-c49920304a00485cbed674dfc0677ec47b04eefcb45cacfbfe25ce6f06004c14R199) _could_ be causing the unexpected behavior.

![Windows animations disabled](https://user-images.githubusercontent.com/5023024/193864535-4bc1bc54-a944-4eea-9daf-de7c46bf6be1.png)

`beta.95` with unexpected behavior:
![loader-animations-reduce](https://user-images.githubusercontent.com/5023024/193864665-41902c3a-8bd0-4124-82ad-d0c984953558.gif)

Prior to fix in `beta.86` with previous animation:
![loader-animations-reduce-beta86](https://user-images.githubusercontent.com/5023024/193864679-e35bf795-2873-46ca-b0c0-ce1172f75cd8.gif)

**Comment by geospatialem** [comment]
Blocked by #5206.

Since the duration cannot be set to `0`, the loader's animation is trying to finish in 3ms, which is way too quick. 

We can either:
1. Hold off on 5206 to land so that this is fixed, or 
2. As loader is not a `openClose` component yet, we could set it internally inside the loader host and remove after the 5206 fix.

cc @anveshmekala

**Comment by jkieboom** [comment]
Same experience as mentioned in https://github.com/Esri/calcite-components/issues/3489#issuecomment-1267211343 on macos with the "Reduce motion" option enabled through Preferences -> Accessibility -> Display

**Comment by geospatialem** [comment]
Verified on Windows and Mac OS when reduced motion is enabled in `next.672` with [the following Codepen](https://codepen.io/geospatialem/pen/KKejmVy).

![image](https://user-images.githubusercontent.com/5023024/207403415-330a69c0-1e09-4459-ac66-dfaff1a7acfc.png)
